File: leetcode/2525.根据规则将箱子分类.py
class Solution:
    def categorizeBox(self, length, width, height, mass):
        maxd = max(length, width, height)
        vol = length * width * height
        isBulky = maxd >= 10000 or vol >= 10**9
        isHeavy = mass >= 100
        if isBulky and isHeavy:
            return 'Both'
        if isBulky:
            return 'Bulky'
        if isHeavy:
            return 'Heavy'
        return 'Neither'

File: leetcode/2525.根据规则将箱子分类.go
func categorizeBox(length, width, height, mass int) string {
    maxd := max(length, max(width, height))
    vol := length * width * height
    isBulky := maxd >= 10000 || vol >= 1e9
    isHeavy := mass >= 100
    if isBulky && isHeavy {
        return "Both"
    } else if isBulky {
        return "Bulky"
    } else if isHeavy {
        return "Heavy"
    } else {
        return "Neither"
    }
}

func max(a, b int) int {
  if a > b {
      return a
  }
  return b
}